src/compiler/proposals-and-stabilization/custom-test-jobs.md

```diff
@@ -0,0 +1,96 @@
+# Adding custom test jobs or adding custom test components to rust-lang/rust CI
+
+## Scope
+
+This policy is applicable to proposals for adding new test jobs or adjusting existing test jobs to build additional custom things that are *not* shipped to the end users, but may still cause [rust-lang/rust] PR CI or Full CI (the "CI") to fail, or produce a failure message, which may impact compiler/library/tools contributors who also use the [rust-lang/rust] CI.
+
+For example, this includes but is not limited to:
+
+- Ecosystem test jobs, such as Rust for Linux or Fuchsia;
+- Custom codegen backend test components, such as `codegen_cranelift` or `codegen_gcc`, which are not part of the "main" distributed toolchain.
+
+## Background
+
+[rust-lang/rust] runs a small set of test (and build) jobs on PR CI (faster, less expensive/exhaustive), and runs a much larger set of test (and build) jobs on Full CI. PR CI usually takes around an hour, while Full CI usually takes around 3 hours. Having test jobs that:
+
+- may spuriously fail; or
+- may genuinely fail but it's not clear who should be consulted about the failure or who is responsible for fixing the failure; or
+- otherwise do not have well documented test job maintainers and failure protocol;
+
+can introduce a lot of friction and frustration for other contributors who now will have to deal with the failure or understand why the test job failed in the first place, which may now be blocking their PR. Everyone utilizing the [rust-lang/rust] CI is expected to use it responsibly.
+
+To help with this, please follow the process described below if you'd like to propose adding a custom test job/component to the [rust-lang/rust] CI, and copy and fill out the Custom Test Job/Component Policy.
+
+## Process for adding a custom test job to rust-lang/rust CI
+
+![Flowchart of the process for adding a custom test job/component to rust-lang/rust CI](./custom-test-jobs/test-job-flow.png)
```

Comment:
I don't suppose that we could build this e.g. with Mermaid or at least SVG? Having this be a PNG (without some source) essentially means that it won't be ever possible to be changed without redoing it from scratch :)

Reply:
It can be, this is mostly *a* diagram. Happy to change it to whatever format that's more maintainble, I just used a naive draw.io flowchart for illustrative purposes.

Reply:
The exported SVG form is

![test-job-flow drawio](https://github.com/user-attachments/assets/2a6634e3-dd06-4973-b5bd-900d665026b7)

but yeah.

Reply:
We could just commit the SVG, I think. Could be rendered a bit differently on different browsers, but that shouldn't be such an issue here.

Reply:
Yeah, that seems at least easier to edit than PNG, although I can't say I'm an expert at editing an SVG either.

Reply:
Committed the SVG instead of the PNG.